plugins/woocommerce/includes/legacy/class-wc-legacy-api.php

```diff
@@ -62,6 +68,56 @@ public function add_query_vars( $vars ) {
 		return $vars;
 	}
 
+	/**
+	 * Write a log entry and update the last usage options, for a Legacy REST API request.
+	 *
+	 * @param string      $route The Legacy REST API route requested.
+	 * @param string|null $user_agent The content of the user agent HTTP header in the request, null if not available.
+	 */
+	private function maybe_log_rest_api_request( string $route, ?string $user_agent ) {
+		if ( is_plugin_active( 'woocommerce-legacy-rest-api/woocommerce-legacy-rest-api.php' ) ) {
+			return;
+		}
+
+		$user_agent = $user_agent ?? 'unknown';
+
+		$current_date        = wp_date( 'Y-m-d H:i:s' );
+		$stored_api_accesses = get_transient( 'wc_legacy_rest_api_usages' );
+		if ( false === $stored_api_accesses ) {
+			$stored_api_accesses = array(
+				'user_agents' => array(),
+				'first_usage' => $current_date,
+				'total_count' => 0,
+			);
+		}
+
+		$stored_api_accesses_for_user_agent = $stored_api_accesses['user_agents'][ $user_agent ] ?? null;
+		if ( is_null( $stored_api_accesses_for_user_agent ) ) {
+			$stored_api_accesses['user_agents'][ $user_agent ] = array(
+				'first_date' => $current_date,
+				'last_date'  => $current_date,
+				'count'      => 1,
+			);
+		} else {
+			$stored_api_accesses['user_agents'][ $user_agent ]['count']++;
+			$stored_api_accesses['user_agents'][ $user_agent ]['last_date'] = $current_date;
+		}
+		$stored_api_accesses['total_count']++;
+
+		set_transient( 'wc_legacy_rest_api_usages', $stored_api_accesses, time() + 2 * WEEK_IN_SECONDS );
+
+		/**
+		 * This filter allows disabling the logging of Legacy REST API usages.
+		 *
+		 * @param bool $do_logging True to enable the logging of all the Legacy REST API usages (default), false to disable.
+		 *
+		 * @since 8.5.0
+		 */
+		if ( apply_filters( 'woocommerce_log_legacy_rest_api_usages', true ) ) {
+			wc_get_logger()->info( 'LEGACY REST API USAGE DETECTED (version ' . WC_API_REQUEST_VERSION . "): $route (User agent: $user_agent)", array( 'source' => 'legacy_rest_api_usages' ) );
```

Comment:
There's obviously nothing wrong with this format, but maybe we could one that would be "easier" to parse or tabulate?
For example instead of

```
2023-12-19T13:34:21+00:00 INFO LEGACY REST API USAGE DETECTED (version 3): /orders (User agent: RapidAPI/4.2.0 (Macintosh; OS X/14.2.0) GCDHTTPRequest)
```

We could log something like

```
2023-12-19T13:34:21+00:00 INFO Version: 3, Route: /orders, User agent: RapidAPI/4.2.0 (Macintosh; OS X/14.2.0) GCDHTTPRequest
```

Not that it'd make a _ton_ of difference but it'd be slightly easier to parse (in case merchants want to export to Excel or something). Also since this is already inside the legacy usages log, maybe we can drop the "LEGACY REST API USAGE DETECTED" bit as it could be considered redundant.